validphys2/src/validphys/photon/compute.py

```diff
@@ -313,34 +412,67 @@ def set_thresholds_alpha_em(self):
+        # determine the values of the couplings in the threshold points, depending on the value of qref
         for nf in range(nfref + 1, self.theory["MaxNfAs"] + 1):
-            alpha_thresh[nf] = self.alpha_em_fixed_flavor(
-                thresh[nf], alpha_thresh[nf - 1], thresh[nf - 1], nf - 1
+            coupl_tmp = self.couplings_fixed_flavor(
+                thresh[nf], couplings_thresh[nf - 1], thresh[nf - 1], nf - 1
             )
+            # compute matchings
+            match_up = compute_matching_coeffs_up(self.theory["HQ"], nf)
+            alpha_s = coupl_tmp[0]
+            alphas_new = apply_match(alpha_s, self.theory['PTO'] + 1, logs[nf], match_up)
+            couplings_thresh[nf] = [alphas_new, coupl_tmp[1]]
 
         for nf in reversed(range(3, nfref)):
-            alpha_thresh[nf] = self.alpha_em_fixed_flavor(
-                thresh[nf], alpha_thresh[nf + 1], thresh[nf + 1], nf + 1
+            coupl_tmp = self.couplings_fixed_flavor(
+                thresh[nf], couplings_thresh[nf + 1], thresh[nf + 1], nf + 1
             )
-
-        return thresh, alpha_thresh
-
-    def set_betas(self):
-        """Compute and store beta0 / 4pi and b1 = (beta1/beta0)/4pi as a function of nf."""
-        beta0 = {}
-        b1 = {}
+            # compute matchings
+            match_down = compute_matching_coeffs_down(self.theory["HQ"], nf - 1)
+            alpha_s = coupl_tmp[0]
+            alphas_new = apply_match(alpha_s, self.theory['PTO'] + 1, logs[nf + 1], match_down)
+            couplings_thresh[nf] = [alphas_new, coupl_tmp[1]]
+
+        return thresh, couplings_thresh
+
+    def compute_betas(self):
+        """Set values of betaQCD and betaQED"""
+        betas_qcd = {}
+        betas_qed = {}
+        beta_mix_qcd = {}
+        beta_mix_qed = {}
         for nf in range(3, 6 + 1):
-            nu = nf // 2
-            nd = nf - nu
-            beta0[nf] = (-4.0 / 3 * (NL + NC * (nu * EU2 + nd * ED2))) / (4 * np.pi)
-            b1[nf] = (
-                -4.0 * (NL + NC * (nu * EU2**2 + nd * ED2**2)) / beta0[nf] / (4 * np.pi) ** 2
-            )
-        return beta0, b1
+            vec_qcd = [beta.beta_qcd_as2(nf) / (4 * np.pi)]
+            vec_qed = [beta.beta_qed_aem2(nf) / (4 * np.pi)]
+            for ord in range(1, self.theory['PTO'] + 1):
+                vec_qcd.append(beta.b_qcd((ord + 2, 0), nf) / (4 * np.pi) ** ord)
+            for ord in range(1, self.theory['QED']):
+                vec_qed.append(beta.b_qed((0, ord + 2), nf) / (4 * np.pi) ** ord)
+            betas_qcd[nf] = vec_qcd
+            betas_qed[nf] = vec_qed
+            beta_mix_qcd[nf] = beta.b_qcd((2, 1), nf) / (4 * np.pi)
+            beta_mix_qed[nf] = beta.b_qed((1, 2), nf) / (4 * np.pi)
+        return betas_qcd, betas_qed, beta_mix_qcd, beta_mix_qed
+
+
+def apply_match(alphas, ord, L, match):
+    "applying matching conditions to alphas"
```

Comment:
```suggestion
    """Apply the matching conditions to alpha_s"""
```

Please, add more information to the docstr on what the input is.